presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -427,7 +427,9 @@ public static Object getField(Type type, Block block, int position)
 
     public static void checkTableIsWritable(Table table, boolean writesToNonManagedTablesEnabled)
     {
-        if (!writesToNonManagedTablesEnabled && !table.getTableType().equals(MANAGED_TABLE.toString())) {
+        if (!writesToNonManagedTablesEnabled
+                && !table.getTableType().equals(MANAGED_TABLE.toString())
+                && !table.isTemporary()) {
             throw new PrestoException(NOT_SUPPORTED, "Cannot write to non-managed Hive table");
```

Comment:
nit: Make the exception message also cover temp table.

Reply:
Currently the temporary table concept should be opaque for the end user. I'm not sure if it makes sense to mention the temporary tables in the end user message.